### ISSUE TITLE
0.5.0 release notes

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#879]
A start on 0.5.0 release notes, based on merge commits emitted from this script (with one multi-line description fixed up by hand).  It'd be nice if we could just start using merge commit descriptions directly in NEWS.md.  What do you think, too verbose?

```sh
#!/bin/bash
LASTTAG=v0.4.1
echo Changes since $LASTTAG:
git log --pretty='format:%b' --merges $LASTTAG..HEAD | while read line; do
	echo "* $line"
done
```

Changes since v0.4.1:
* fix MANPATH for Ubuntu, and tidy up travis dep builder
* fixes for minor issues detected by Coverity
* build: add --disable-docs configure option
* kvs: allow get_double to be called on an int
* README.md: Update srun instructions
* travis-ci: minor updates
* misc minor fixes
* make flux_msg_t a bonafide type, add jansson payload accessors
* Fix --rank issues, add NODESET documentation, and minor cleanup
* Fix output errors with flux up --comma & --newline, add appropriate tests
* Add hierarchical lwj directory support in kvs
* jsc test: Add timed waits to avoid races
* t/t0005-exec.t: Fix corner case in test for file not found
* doc/man1/flux-start.adoc: Fix example option usage
* add dlopen RTLD_DEEPBIND flag
* src/broker/broker.c: Fix typo flux_repond -> flux_respond
* doc/man1/flux-module.adoc: Fix environment variable error
* Fix make distcheck
* Pull in json-c, allowing internals to link against alternate json libraries.
Add enhanced flux_rpc functions using libjansson json_pack/unpack functions
* Update flux_t * references in man pages
* Remove pointer from typedef flux_t
* t/t2000-wreck.t: Fix invalid compare on per-task affinity test
* t/t2000-wreck.t: Fix invalid compare on 'wreckrun: --input=0 works' test
* Remove JSON typedef, just use json_object *
* module: Remove pointer from typedef flux_modlist_t
* security: Remove pointer from typedef flux_sec_t
* kvs: add kvs_getat() and related functions
* experimental aggregator module
* kvs: testing, fix use-after-free, streamline python bindings
* Fix #821: crash in kvs due to NULL arg in Jget_str()
* python: add a check for invalid handle types
* Python json and constant rework
* Python destructor refactoring and exception safety
* libutil/veb: quiet uninitialized variable warning in vebnew
* when tagpool is exhausted, grow up to RFC 6 limits
* add KVS blobref access functions
* Fix missing error checks in Lua bindings, flux-wreckrun, flux-wreck
* python: Several fixes for the bindings
* Switch lua scripts to use lua interpreter in PATH
* restructure kvs commit handling code for correctness
* broker/hello: fix leak/error detection in flux_rpc
* implement scalable reduction for wireup protocol
* wreck: minor enhancements for scale testing
* increase KVS commit window
* autogen.sh: run libtoolize before autoreconf
* clean up LOG_INFO output, log wireup, rc1, rc3 times, add pmi timing.
* optparse: remove requirement for option key on long-only options (and other fixes)
* travis.yml:  Fix ANCHOR definition

**Comment by garlick** [comment]
Or slightly better?
```sh
#!/bin/bash
LASTTAG=v0.4.1
echo Changes since $LASTTAG:
git log --pretty='format:%s%n%b' --merges $LASTTAG..HEAD | while read line; do
	if echo $line|grep -q "Merge pull"; then
		pr=$(echo $line|sed -e 's/Merge pull request //' -e 's/ from .*//')
	else
		echo "* $pr: $line"
	fi
done
```
Changes since v0.4.1:
* #877: fix MANPATH for Ubuntu, and tidy up travis dep builder
* #876: fixes for minor issues detected by Coverity
* #871: build: add --disable-docs configure option
* #872: kvs: allow get_double to be called on an int
* #867: README.md: Update srun instructions
* #865: travis-ci: minor updates
* #862: misc minor fixes
* #857: make flux_msg_t a bonafide type, add jansson payload accessors
* #860: Fix --rank issues, add NODESET documentation, and minor cleanup
* #858: Fix output errors with flux up --comma & --newline, add appropriate tests
* #811: Add hierarchical lwj directory support in kvs
* #859: jsc test: Add timed waits to avoid races
* #848: t/t0005-exec.t: Fix corner case in test for file not found
* #852: doc/man1/flux-start.adoc: Fix example option usage
* #849: add dlopen RTLD_DEEPBIND flag
* #851: src/broker/broker.c: Fix typo flux_repond -> flux_respond
* #850: doc/man1/flux-module.adoc: Fix environment variable error
* #847: Fix make distcheck
* #835: Pull in json-c, allowing internals to link against alternate json libraries.
* #835: Add enhanced flux_rpc functions using libjansson json_pack/unpack functions
* #844: Update flux_t * references in man pages
* #841: Remove pointer from typedef flux_t
* #837: t/t2000-wreck.t: Fix invalid compare on per-task affinity test
* #836: t/t2000-wreck.t: Fix invalid compare on 'wreckrun: --input=0 works' test
* #832: Remove JSON typedef, just use json_object *
* #831: module: Remove pointer from typedef flux_modlist_t
* #830: security: Remove pointer from typedef flux_sec_t
* #824: kvs: add kvs_getat() and related functions
* #787: experimental aggregator module
* #823: kvs: testing, fix use-after-free, streamline python bindings
* #822: Fix #821: crash in kvs due to NULL arg in Jget_str()
* #819: python: add a check for invalid handle types
* #799: Python json and constant rework
* #807: Python destructor refactoring and exception safety
* #809: libutil/veb: quiet uninitialized variable warning in vebnew
* #806: when tagpool is exhausted, grow up to RFC 6 limits
* #801: add KVS blobref access functions
* #804: Fix missing error checks in Lua bindings, flux-wreckrun, flux-wreck
* #794: python: Several fixes for the bindings
* #789: Switch lua scripts to use lua interpreter in PATH
* #788: restructure kvs commit handling code for correctness
* #786: broker/hello: fix leak/error detection in flux_rpc
* #781: implement scalable reduction for wireup protocol
* #782: wreck: minor enhancements for scale testing
* #780: increase KVS commit window
* #771: autogen.sh: run libtoolize before autoreconf
* #769: clean up LOG_INFO output, log wireup, rc1, rc3 times, add pmi timing.
* #768: optparse: remove requirement for option key on long-only options (and other fixes)
* #767: travis.yml:  Fix ANCHOR definition

**Comment by grondo** [comment]
That's pretty good, thanks! I wonder if for official release notes we can prune testing only fixes, or if we can just author very high level bullets for user release notes (maybe that is more future work, for now this seems fine).

**Comment by garlick** [comment]
Last time we manually sorted the items by topic.  Maybe rather than prune testing only fixes, we could move them to a testing category?

**Comment by grondo** [comment]
Oh yeah, that sounds reasonable. BTW, for the script (really nice, btw) you can get the most recent tag on the current branch with `LASTTAG=$(git describe --tags --abbrev=0)`

**Comment by chu11** [comment]
> I wonder if for official release notes we can prune testing only fixes, or if we can just author very high level bullets for user release notes (maybe that is more future work, for now this seems fine).

I like what I've seen in a number of Apache projects.  Release notes for revision number increase releases just get a list of jira issue numbers and a sentence ripped out of jira.  Minor number increase releases get an additional high level summary of important updates.  I like it, as the former is only bug fixes so the list of jiras is sufficient.  The minor release must have something non-trivial or non-bug-fix, thus warranting the summary paragraph.

Obviously we're pre-1.0, so maybe we don't care right now.  But FWIW

**Comment by grondo** [comment]
@chu11, Yeah somewhere we have an issue (I think) describing a possible format for PR commit messages which could allow Release Notes to be built automatically. E.g. we could add a "topic" tag so something like @garlick's script could automatically organize by topic.

**Comment by garlick** [comment]
I can't find that issue (here or in distribution project) - maybe we didn't open one?  @chu11 feel free to open one regarding post 1.0 release formats and link an example of what you like.

**Comment by garlick** [comment]
We don't have any other issues linked to the 0.5.0 milestone.  Do we have any blockers or should I go ahead and submit a PR for the release notes?

**Comment by grondo** [comment]
I kind of wanted to do a sanity check at scale but I've been having connectivity issues here at home so didn't get to it today. I can't think of any other blockers, but we should test build an RPM and ensure the release can successfully be used with flux-sched...

**Comment by garlick** [comment]
I sent ahead and opened up issues for those activities and linked them to the milestone.  I'll tentatively prepare release notes but we shouldn't merge that PR until we're gone through those.

**Comment by grondo** [comment]
Thanks!

**Comment by chu11** [comment]
@garlick Sure thing, I'll open up an issue in distribution as a discussion starter.
